File: daggerandroid/src/main/java/com/daggerandroid/sample/di/module/MainActivityModule.java
package com.daggerandroid.sample.di.module;

import com.daggerandroid.library.di.scope.ActivityScope;
import com.daggerandroid.sample.MainActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Copyright (c), 2018-2019
 *
 * @author: lixin
 * Date: 2019/4/12
 * Description:
 */
//@Module(subcomponents = {MainActivitySubCompoent.class})
@Module
public abstract class MainActivityModule {

//    @Binds
//    @IntoMap
//    @ActivityKey(MainActivity.class)
//    abstract AndroidInjector.Factory<? extends Activity>
//    bindActivityInjectorFactory(MainActivitySubCompoent.Builder builder);

    @ActivityScope
    @ContributesAndroidInjector(modules = StudentModule.class)
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/dagger2/sample/di/modules/MainActivityModule.kt
package com.dagger2.sample.di.modules

import com.dagger2.sample.di.annotation.MainActivityScope
import com.dagger2.sample.ui.MainActivity
import dagger.Module
import dagger.Provides

/**
 * Copyright (c), 2018-2019
 * @author: lixin
 * Date: 2019/4/3
 * Description:
 */
@Module
class MainActivityModule(private val mainActivity: MainActivity) {

    @Provides
    @MainActivityScope
    fun mainActivity(): MainActivity {
        return mainActivity
    }
}